parse_grid.py

```
@@ -40,6 +40,14 @@ def safe_find_all(element: Union[Tag, BeautifulSoup], name: str, **kwargs) -> Li
     return [r for r in results if isinstance(r, Tag)]
 
 
+def get_soup(html_content: str) -> BeautifulSoup:
+    """Get a BeautifulSoup object, with lxml speed but html.parser fallback."""
+    try:
+        return BeautifulSoup(html_content, "lxml")
+    except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)